barretenberg/cpp/src/barretenberg/vm2/simulation/events/alu_event.hpp

```diff
@@ -12,11 +12,18 @@ enum class AluOperation {
     ADD,
 };
 
+// TODO(MW): Expand when adding new ops (e.g. when using max_bits for mul, we would cover bits related errors)
+enum class AluError {
+    // TODO(MW): Split into cases i.e. ab tags not equal, c tag not as expected, ..., ?
+    TAG_ERROR,
+};
+
 struct AluEvent {
     AluOperation operation;
     MemoryValue a;
     MemoryValue b;
     MemoryValue c;
+    std::optional<AluError> error;
```

Comment:
For now, I used a simple approach and optionally emit an error which currently only has one case. Probably in the future I will make `c` a struct with something like `.value` and `.error` to avoid emitting events with invalid `c` results?